### PR TITLE
Discuss optional intermediate cert for private root CA

### PR DESCRIPTION
**Pull request by sean-horn**
Not a commonly used config, but possible, so let's doc it

Signed-off-by: Sean Horn <sean_horn@chef.io>

### CODE REVIEW

File: chef_master/source/server_security.rst [hunk 1]
```diff
@@ -226,14 +226,22 @@ Private Certificate Authority
...
 
-To use an internal certificate authority, append both the server and root certificates into a single ``.crt`` file. For example:
+To use an internal certificate authority, append both the server (optionally, any intermediate cert as well) and root certificates into a single ``.crt`` file. For example:
 
 .. code-block:: bash
 
-   $ cat server.crt root.crt >> /var/opt/opscode/nginx/ca/FQDN.crt
+   $ cat server.crt [intermediate.crt] root.crt >> /var/opt/opscode/nginx/ca/FQDN.crt
+
+You can check the validity of your new combined cert like this, on the Chef Server. The cacert.pem only needs to contain your root CA's cert file. This is not the usual treatment, but mimics how ChefDK behaves after a ``knife ssl fetch`` followed by a ``knife ssl verify``
```
Comment (mjingle) [inline chef_master/source/server_security.rst]: @sean-horn Thank you for writing this up! Can you clarify which code block (line 233 or line 237) you're referring to when you say "like this" in line 235?

Comment (sean-horn) [reply chef_master/source/server_security.rst]: @mjingle I put the discussion before the code blocks here, so line 235 is referring to line 237.

File: chef_master/source/server_security.rst [hunk 2]
```diff
@@ -226,14 +226,22 @@ Private Certificate Authority
...
 
-To use an internal certificate authority, append both the server and root certificates into a single ``.crt`` file. For example:
+To use an internal certificate authority, append the server--optionally, any intermediate certificate as well--and root certificates into a single ``.crt`` file. For example:
 
 .. code-block:: bash
 
-   $ cat server.crt root.crt >> /var/opt/opscode/nginx/ca/FQDN.crt
+   $ cat server.crt [intermediate.crt] root.crt >> /var/opt/opscode/nginx/ca/FQDN.crt
+
+You can check the validity of your new combined certificate on the Chef Server, using the command below. The cacert.pem only needs to contain your root CA's certificate file. This is not the usual treatment, but mimics how ChefDK behaves after a ``knife ssl fetch`` followed by a ``knife ssl verify``.
```
Comment (kagarmoe) [inline chef_master/source/server_security.rst]: Check your combined certificate's validity  on the Chef Server: 

	.. code-block:: bash
    $ openssl verify -verbose -purpose sslserver -CAfile cacert.pem  /var/opt/opscode/nginx/ca/FQDN.crt 

The cacert.pem should contain only your root CA's certificate file. This is not the usual treatment, but mimics how ChefDK behaves after a ``knife ssl fetch`` followed by a ``knife ssl verify``.

**Review by mjingle — APPROVED** [review]
tenor-146710396

**Pull request merged**